### ISSUE TITLE
Feature request: Client only option

### ISSUE THREAD

**Issue by POMATu** [yggdrasil-network/yggdrasil-go#1068]
So I am having few VPSes that i am using to contribute to network, my home broadband ISP is also good so I can contribute there too but I am also having laptop, smartphone and other devices like that where internet can be occasionally very bad (some wifi, metered 4g and so on).

I2P has such option so why does yggdrasil doesnt have it? I want to disable relaying other ppl traffic completely on such client-only devices. Is it even possible with yggdrasil design?

Yes traffic might never get relayed via me because yggdrasil is taking shortcuts, but sometimes it looks like it does, and why do other people need to get this shit latency when I know that internet is shit on these particular devices and I have better devices to contribute to network? And thats me, other people might just plug bunch of nodes inside their yggdrasil config and start relaying my traffic via their shit connection which i also dont want.

As temporary solution I am only using my VPSes as entry nodes this way traffic doesnt get routed to my low quality devices because this would be too roundabout path still I think we need an option for client-only mode, and while some people might abuse it and disable relaying without any sense init i think the actual yggdrasil pathes and latency will get better this way when people can choose which of their devices are 24/7 relays and which are just occasional clients.

Thats if its even possible with mesh

**Comment by zhoreeq** [comment]
Since Yggdrasil is a mesh network, every node should route packets equally.

In your case, it'd be much more efficient to set up a Wireguard server for "light" clients with access to Yggdrasil. Just assign 300::/8 addresses to clients.

**Comment by POMATu** [comment]
yh thats if wireguard protocol is not DPI'ed
and also wireguard is one single connection to VPS if it goes down then its bottleneck, on android you can only have 1 VPN per profile
whole point of yggdrasil is that it can have a lot of connections to different servers

so what you propose is working but its half-assed

**Comment by zhoreeq** [comment]
>whole point of yggdrasil is that it can have a lot of connections to different servers

Not really. Ideally, there should be no Internet peers at all, only local peers over WiFi and LAN.

**Comment by POMATu** [comment]
> Ideally

You are living in some other ideal world, i am living in reality with 50-100 public peers listed on github. Surely there gonna be local yggdrasil peers on 4G connection (no)
